### PR TITLE
user mapping and manual mapping changes

### PR DESCRIPTION
**Pull request by pooja-koul-incontact**

### CODE REVIEW

File: src/app/user-mapping/user-mapping.component.ts [hunk 1]
```diff
@@ -26,7 +27,7 @@ export class UserMappingComponent implements OnInit {
   selectedRows: any[];
   isMapped = false;
   isSaved = false;
-  results: UserMappingSearchUserData[];
+  searchResults: UserMappingSearchUserData[];
   response: any;
   message: string;
   found: boolean;
```
Comment (abhilash-incontact) [inline src/app/user-mapping/user-mapping.component.ts]: can you be more specific like isUserFound or something?

File: src/app/user-mapping/user-mapping.component.ts [hunk 2]
```diff
@@ -26,7 +27,7 @@ export class UserMappingComponent implements OnInit {
   selectedRows: any[];
   isMapped = false;
   isSaved = false;
-  results: UserMappingSearchUserData[];
+  searchResults: UserMappingSearchUserData[];
   response: any;
```
Comment (abhilash-incontact) [inline src/app/user-mapping/user-mapping.component.ts]: Why this is any?

File: src/app/user-mapping/user-mapping.component.spec.ts [hunk 3]
```diff
@@ -69,6 +69,19 @@ describe('UserMappingComponent', () => {
...
     expect(component.autoMapUsers).toBeTruthy();
   });
+  it('should call autoSearch method', () => {
+    const searchUser: UserMappingSearchUserData[] = null;
+    const service: UserMappingDataService = TestBed.get(UserMappingDataService);
+    spyOn(service, 'autoSearch').and.returnValue(of(searchUser));
+    component.queryField.valueChanges.subscribe(query => service.autoSearch('abc', '123')
+      .subscribe((response: UserMappingSearchUserData[]) => {
+        if (response.length >= 1) {
+          expect(component.found === true);
```
Comment (abhilash-incontact) [inline src/app/user-mapping/user-mapping.component.spec.ts]: You don't need to evaluate boolean expressions. A cleaner way would be simply use expect(component.found). Do this all tests.

File: src/app/user-mapping/user-mapping.component.html [hunk 4]
```diff
@@ -78,25 +78,28 @@ <h3>Search Users</h3>
...
         <lib-grid-v7 [columnDefs]="mappedUsersColumnDefs" [rowData]="mappedUsersRowData" (selected_data)='getSelectedRows($event)' ></lib-grid-v7>
     </div>
-  </div>
 
-  <div id="unmappedUsers" class ="row grid-container tab-pane fade" *ngIf="showUnMapped">
-    <div class="col-md-12">
-      <lib-grid-v7 [columnDefs]="unmappedUsersColumnDefs" [rowData]="unmappedUsersRowData" (map_user)='unmappedUsersSearch()' data-target="#searchUser" data-toggle="modal"></lib-grid-v7>
+
+    <div class="col-md-12" class="tab-pane" id="unmappedUsers" *ngIf="showUnMapped">
+      <!--<lib-grid-v7 [columnDefs]="unmappedUsersColumnDefs" [rowData]="unmappedUsersRowData" (map_user)='unmappedUsersSearch()' data-target="#searchUser" data-toggle="modal"></lib-grid-v7>-->
```
Comment (abhilash-incontact) [inline src/app/user-mapping/user-mapping.component.html]: Remove commented code

File: src/app/user-mapping/services/user-mapping-data.service.ts [hunk 5]
```diff
@@ -7,6 +7,10 @@ import {UserMappingComponent} from '../user-mapping.component';
   providedIn: 'root'
 })
 export class UserMappingDataService {
+  firstName: string;
+  lastName: string;
+  email: string;
+  buId: string;
 
   constructor(private http: HttpClient) { }
   getRefreshUserData(buMappingId) {
```
Comment (abhilash-incontact) [inline src/app/user-mapping/services/user-mapping-data.service.ts]: why we have hardcoded values here?

File: src/app/user-mapping/manual-mapping/manual-mapping.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (abhilash-incontact) [inline src/app/user-mapping/manual-mapping/manual-mapping.component.spec.ts]: Add more test cases

File: src/app/user-mapping/services/user-mapping-data.service.ts (hunk 5 above)
Comment (pooja-koul-incontact) [reply src/app/user-mapping/services/user-mapping-data.service.ts]: here I am just capturing the values from api here, nothing is hardcoded

File: src/app/user-mapping/manual-mapping/manual-mapping.component.spec.ts (hunk 6 above)
Comment (pooja-koul-incontact) [reply src/app/user-mapping/manual-mapping/manual-mapping.component.spec.ts]: this screen is still under development, just added some code for the demo purpose, will be adding the test cases soon.

File: src/app/app.module.ts [hunk 7]
```diff
@@ -56,7 +58,9 @@ import { LoadingSpinnerComponent } from './bu/loading-spinner/loading-spinner.co
     ClickIncontactBuIdComponent,
     ClickPartnerBuIdComponent,
     ClickPartnerIdComponent,
-    LoadingSpinnerComponent
+    LoadingSpinnerComponent,
+    ClickPartnerIdComponent,
```
Comment (saiprasad1997) [inline src/app/app.module.ts]: Click Partner Id Component is repeated

Comment (pooja-koul-incontact) [reply src/app/app.module.ts]: removed. Please check

**Pull request merged**